### PR TITLE
Coverity fixes 20220714

### PR DESCRIPTION
**Pull request by paulidale**
Fixing Coverity 1506566, 1503326, 1503321 & 1503327.

### CODE REVIEW

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by paulidale** [comment]
The various `_is_a` changes will cascade further but not here not now.

**Review by hlandau — APPROVED** [review]

File: crypto/bio/bss_dgram.c [hunk 1]
```diff
@@ -1914,7 +1914,8 @@ static void get_current_time(struct timeval *t)
     t->tv_sec = (long)(now_ul / 10000000);
     t->tv_usec = ((int)(now_ul % 10000000)) / 10;
 # else
-    gettimeofday(t, NULL);
+    if (gettimeofday(t, NULL) < 0)
+        perror("gettimeofday");
```
Comment (t8m) [inline crypto/bio/bss_dgram.c:1918]: How ugly! I see the whole bss_dgram using perror() quite liberally. Something to refactor?

**Review by t8m — APPROVED** [review]

File: crypto/bio/bss_dgram.c (hunk 1 above)
Comment (paulidale) [reply crypto/bio/bss_dgram.c:1918]: We should.  The number of `void` functions used makes this painful -- they cannot return an error indication and this seems (slightly) better than having asserts.

**Comment by paulidale** [comment]
Merged, thanks for the reviews.

**Pull request closed without merging**